modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java

```diff
@@ -498,4 +502,16 @@ public void execute(Template.Fragment fragment, Writer writer) throws IOExceptio
             writer.write(fragment.execute().replaceAll(from, to));
         }
     }
+
+    // Can't figure out the logic in DefaultCodegen but optional vars are getting duplicated when there's
```

Comment:
Agreed with you it should be done in DefautlCodegen. I'll try to figure out a fix later in DefaultCodegen so as to fix the issue for other generators as well.